Add noble packaging support in fw_icu4c_ci.yml

jasonleenaylor opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/icu/24)
<!-- Reviewable:end -->

jasonleenaylor committed:
Update fw_icu4c_ci.yml

jasonleenaylor committed:
Update GitHub Actions to latest versions

jasonleenaylor committed:
Add noble in a couple more places

ermshiperete commented on .github/workflows/fw_icu4c_ci.yml:
```diff
@@ -161,25 +161,25 @@ jobs:
           version="${{needs.set-version-number.outputs.version}}${{ env.package_suffix }}"
           ${CI_SCRIPT_DIR}/bash/make-source \
-            --dists "xenial focal bionic" \
+            --dists "noble xenial focal bionic" \
```
We should remove the EOL versions (xenial, bionic) and focal which will be EOL in April.

```suggestion
            --dists "noble jammy" \
```

ermshiperete commented on .github/workflows/fw_icu4c_ci.yml:
```diff
@@ -161,25 +161,25 @@ jobs:
             --arches "amd64" \
             --main-package-name "icu-fw" \
-            --supported-distros "xenial focal bionic" \
+            --supported-distros "noble xenial focal bionic" \
```
```suggestion
            --supported-distros "noble jammy" \
```

ermshiperete commented on .github/workflows/fw_icu4c_ci.yml:
```diff
@@ -161,25 +161,25 @@ jobs:
           sudo sbuild-adduser runner
           sg sbuild '${CI_SCRIPT_DIR}/bash/build-package \
-            --dists "xenial focal bionic" \
+            --dists "noble xenial focal bionic" \
```
```suggestion
            --dists "noble jammy" \
```

ermshiperete commented on .github/workflows/fw_icu4c_ci.yml:
```diff
@@ -161,25 +161,25 @@ jobs:
             --arches "amd64" \
             --main-package-name "icu-fw" \
-            --supported-distros "xenial focal bionic" \
+            --supported-distros "noble xenial focal bionic" \
```
```suggestion
            --supported-distros "noble jammy" \
```

ermshiperete reviewed (approved):
LGTM

jasonleenaylor committed:
Update .github/workflows/fw_icu4c_ci.yml

Co-authored-by: Eberhard Beilharz <ermshiperete@users.noreply.github.com>

jasonleenaylor committed:
Update .github/workflows/fw_icu4c_ci.yml

Co-authored-by: Eberhard Beilharz <ermshiperete@users.noreply.github.com>

jasonleenaylor committed:
Update .github/workflows/fw_icu4c_ci.yml

Co-authored-by: Eberhard Beilharz <ermshiperete@users.noreply.github.com>

jasonleenaylor committed:
Update .github/workflows/fw_icu4c_ci.yml

Co-authored-by: Eberhard Beilharz <ermshiperete@users.noreply.github.com>

ermshiperete committed:
chore: remove group password

It seems that with Ubuntu 24.04 the sbuild group has a password set.
This change removes the password so that we can use the `sg sbuild`
command without providing a password.

jasonleenaylor committed:
Add universe in repository to debian packaging step

jasonleenaylor committed:
Update after adding the new repositories

jasonleenaylor committed:
Update after adding in the prerequisites step

ermshiperete commented:
I think https://github.com/sillsdev/ci-builder-scripts/pull/7 will fix the package build failure.

hahn-kev commented:
@jasonleenaylor and @ermshiperete the build is now passing again, could we get this merged in? It's blocking #26
